Analysis/Tasks/PWGCF/dptdptcorrelations.cxx

```diff
@@ -0,0 +1,407 @@
+} // namespace aod
+} // namespace o2
+
+namespace dptdptcorrelations
+{
+/* all this bins have to be make configurable */
+int ptbins = 18;
+float ptlow = 0.2, ptup = 2.0;
+int etabins = 16;
+float etalow = -0.8, etaup = 0.8;
+int zvtxbins = 40;
+float zvtxlow = -10.0, zvtxup = 10.0;
+int phibins = 72;
+std::string fTaskConfigurationString = "PendingToConfigure";
+
+Configurable<int> cfgTrackType{"trktype", 1, "Type of selected tracks: 0 = no selection, 1 = global tracks loose DCA, 2 = global SDD tracks"};
+Configurable<int> cfgTrackOneCharge("trk1charge", 1, "Trakc one charge: 1 = positive, -1 = negative");
+Configurable<int> cfgTrackTwoCharge("trk2charge", -1, "Trakc two charge: 1 = positive, -1 = negative");
+Configurable<bool> cfgProcessPairs("processpairs", true, "Process pairs: false = no, just singles, true = yes, process pairs");
+
+/* while we don't have proper bool columnst */
+uint8_t DPTDPT_TRUE = 1;
+uint8_t DPTDPT_FALSE = 0;
+
+/// \enum SystemType
+/// \brief The type of the system under analysis
+enum SystemType {
+  kNoSystem = 0, ///< no system defined
+  kpp,           ///< **p-p** system
+  kpPb,          ///< **p-Pb** system
+  kPbp,          ///< **Pb-p** system
+  kPbPb,         ///< **Pb-Pb** system
+  kXeXe,         ///< **Xe-Xe** system
+  knSystems      ///< number of handled systems
+};
+
+/* probably this will not work in the context of distributed processes */
+/* we need to revisit it!!!!                                           */
+SystemType fSystem = kNoSystem;
+TH1F* fhCentMultB = nullptr;
+TH1F* fhCentMultA = nullptr;
+TH1F* fhVertexZB = nullptr;
+TH1F* fhVertexZA = nullptr;
+TH1F* fhPtB = nullptr;
+TH1F* fhPtA = nullptr;
+TH1F* fhPtPosB = nullptr;
+TH1F* fhPtPosA = nullptr;
+TH1F* fhPtNegB = nullptr;
+TH1F* fhPtNegA = nullptr;
+
+TH1F* fhEtaB = nullptr;
+TH1F* fhEtaA = nullptr;
+
+TH1F* fhPhiB = nullptr;
+TH1F* fhPhiA = nullptr;
+
+TH2F* fhEtaVsPhiB = nullptr;
+TH2F* fhEtaVsPhiA = nullptr;
+
+TH2F* fhPtVsEtaB = nullptr;
+TH2F* fhPtVsEtaA = nullptr;
+
+SystemType getSystemType()
+{
+  /* we have to figure out how extract the system type */
+  return kPbPb;
+}
+
+bool IsEvtSelected(aod::CollisionEvSelCent const& collision)
+{
+  if (collision.alias()[kINT7]) {
+    if (collision.sel7()) {
+      if (zvtxlow < collision.posZ() and collision.posZ() < zvtxup) {
+        return true;
+      }
+      return false;
+    }
+  }
+  return false;
+}
+
+bool matchTrackType(aod::TrackData const& track)
+{
+  switch (cfgTrackType) {
+    case 1:
+      if (track.isGlobalTrack() != 0)
+        return true;
+      else
+        return false;
+      break;
+    case 2:
+      if (track.isGlobalTrackSDD() != 0)
+        return true;
+      else
+        return false;
+      break;
+    default:
+      return false;
+  }
+}
+
+std::tuple<uint8_t, uint8_t> AcceptTrack(aod::TrackData const& track)
```

Comment:
This may be better implemented with partitions - instead of checking each track you can immediately get groups for each of the cases.

Reply:
As it is now it is only checking track charge but the final implementation will need to consider also PID and, perhaps, other configurations. That's why I used this approach for the time being. If finally every is feasible with partitions I will definitely use it.

Reply:
I see. Then you may consider using 'std::pair' here, instead of tuple.

Reply:
I guess the usage is the same because I latter on use to populate two additional data columns which then I use to produce partitions on the tracks table.

Reply:
By the way, as you can see on the commented L.326, I tried to use these partitions to extract combinations out of them but it was rejected by the compiler and I implemented the traditional loop instead

Reply:
I see, I'll try to check what is the problem with those.